### PR TITLE
upgrade IPv6DualStack feature to beta and turn on by default

### PR DESCRIPTION
**Pull request by khenidak**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Upgrades `IPv6Dualstack` to `Beta` and turns it on by default.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
Upgrades `IPv6Dualstack` to `Beta` and turns it on by default. Clusters new and existing will not be affected until user starting adding secondary pod and service cidrs cli flags as described here: https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/563-dual-stack
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]:  https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/563-dual-stack
```

### CODE REVIEW

**Comment by aojea** [comment]
/assign @thockin @liggitt 
/lgtm

Beta graudation requirements are met:

```
Kubernetes types finalized
CRI types finalized
Pods to support multi-IPs
Nodes to support multi-CIDRs
Service resource supports pods with multi-IP
Kubenet to support multi-IPs
```

In addition, there are several optional presubmits and periodic jobs testing the feature:
- https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20dual,%20master
- https://testgrid.k8s.io/sig-network-dualstack-azure-e2e

We are still adding more automated test and working in stabilizing some of the tests and jobs in some platforms, but kind jobs are stable.

Having it enabled by default will allow us ot get a better signal and detect possible problems earlier.

**Comment by aojea** [comment]
/lgtm cancel

heh, it seems we have some real work to do before enabling.

@khenidak lot of tests and places fail because of the following 

> feature gate "IPv6DualStack" with different spec already exists: {true false BETA}

Ironically, it also seems to break KIND, the controller manager panics

```
2021-02-10T16:50:40.493921602Z stderr F panic: feature gate "IPv6DualStack" with different spec already exists: {true false BETA}
2021-02-10T16:50:40.494042788Z stderr F 
2021-02-10T16:50:40.49406094Z stderr F goroutine 1 [running]:
2021-02-10T16:50:40.494070199Z stderr F k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.Must(...)
2021-02-10T16:50:40.494078717Z stderr F 	staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:171
2021-02-10T16:50:40.494085943Z stderr F k8s.io/kubernetes/vendor/k8s.io/controller-manager/pkg/features/register.init.0()
2021-02-10T16:50:40.494092494Z stderr F 	staging/src/k8s.io/controller-manager/pkg/features/register/register.go:26 +0x6f
````

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -751,7 +751,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	LocalStorageCapacityIsolationFSQuotaMonitoring: {Default: false, PreRelease: featuregate.Alpha},
 	NonPreemptingPriority:                          {Default: true, PreRelease: featuregate.Beta},
 	PodOverhead:                                    {Default: true, PreRelease: featuregate.Beta},
-	IPv6DualStack:                                  {Default: false, PreRelease: featuregate.Alpha},
+	IPv6DualStack:                                  {Default: true, PreRelease: featuregate.Beta},
```
Comment (aramase) [inline pkg/features/kube_features.go:754]: Also, need to update the comment here to beta: https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L473-L476

**Comment by aramase** [comment]
We need to update the feature flag here too: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/controller-manager/pkg/features/kube_features.go#L64

After this the tests should pass

**Comment by aojea** [comment]
> /test pull-kubernetes-bazel-test

@khenidak the failure is legit

> k8s.io/kubernetes/pkg/kubelet/dockershim/network/cni.TestCNIPlugin(0xc0003b5b00)
	pkg/kubelet/dockershim/network/cni/cni_test.go:323 +0x2934

File: cmd/kube-apiserver/app/options/options.go [hunk 2]
```diff
@@ -227,10 +227,9 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 		"of type NodePort, using this as the value of the port. If zero, the Kubernetes master "+
 		"service will be of type ClusterIP.")
 
-	// TODO (khenidak) change documentation as we move IPv6DualStack feature from ALPHA to BETA
 	fs.StringVar(&s.ServiceClusterIPRanges, "service-cluster-ip-range", s.ServiceClusterIPRanges, ""+
 		"A CIDR notation IP range from which to assign service cluster IPs. This must not "+
-		"overlap with any IP ranges assigned to nodes or pods.")
+		"overlap with any IP ranges assigned to nodes or pods. Max of two dual-stack cidrs is allowed.")
```
Comment (thockin) [inline cmd/kube-apiserver/app/options/options.go:232]: nit: "CIDRs"

Should flag help strings change based on whether the gate is enabled?

**Review by thockin — COMMENTED** [review]
Scary scary!  Glad to see this.  Just a small sting nit.

File: pkg/features/kube_features.go (hunk 1 above)
Comment (thockin) [reply pkg/features/kube_features.go:755]: last open issue, I think?

**Comment by aojea** [comment]
/lgtm
Go go go

**Review by thockin — COMMENTED** [review]
w00t!

/lgtm
/approve

**Pull request merged**